File: bst.c
#include<stdio.h>
#include<stdlib.h>
typedef struct node{int data;struct node* left,*right;}NODE;
NODE* create_node(int num,NODE** root)
{
    NODE* newnode = (NODE*)calloc(1,sizeof(NODE));
    newnode->data=num;
    if(*root==NULL)
    {
        *root=newnode;
    }
    else
    {
        NODE* tptr=*root;
        NODE* temp;
        while(tptr!=NULL)
        {
            temp=tptr;
            if(tptr->data > num)
            {
                tptr=tptr->left;
            }
            else 
            {
                tptr=tptr->right;
            }
        }
        if(temp->data < num)
        {
            temp->right = newnode;
        }
        else 
        {
            temp->right = newnode;
        }
    }
}
void inorder(NODE* root)
{
    if(root!=NULL)
    {
        inorder(root->left);
        printf("%d ",root->data);
        inorder(root->right);
    }
}
int main()
{
    int num;
    NODE* root=NULL;
    do
    {
        scanf("%d",&num);
        if(num==-1) break;
        create_node(num,&root);
    }while(num!=-1);
    inorder(root);
}

File: BST.java
import java.util.Scanner;

class TreeNode {
    int data;
    TreeNode left, right;

    public TreeNode(int data) {
        this.data = data;
        this.left = this.right = null;
    }
}

class BinarySearchTree {
    TreeNode root;

    BinarySearchTree() {
        root = null;
    }

    void insert(int key) {
        root = insertRec(root, key);
    }

    TreeNode insertRec(TreeNode root, int key) {
        if (root == null) {
            root = new TreeNode(key);
            return root;
        }

        if (key < root.data) {
            root.left = insertRec(root.left, key);
        } else if (key > root.data) {
            root.right = insertRec(root.right, key);
        }

        return root;
    }

    void inorder() {
        System.out.print("Inorder Traversal: ");
        inorderRec(root);
        System.out.println();
    }

    void inorderRec(TreeNode root) {
        if (root != null) {
            inorderRec(root.left);
            System.out.print(root.data + " ");
            inorderRec(root.right);
        }
    }
}
public class BST
{
    public static void main(String[] args) 
    {
        Scanner scanner = new Scanner(System.in);
        BinarySearchTree bst = new BinarySearchTree();

        System.out.println("Enter elements to insert into the Binary Search Tree (enter -1 to stop):");

        int num;
        do {
            num = scanner.nextInt();
            if (num != -1) {
                bst.insert(num);
            }
        } while (num != -1);

        System.out.println("Binary Search Tree created successfully!");
        bst.inorder();

        scanner.close();
    }
}
